pkg/cmd/release/update_releases.go

```diff
@@ -80,10 +82,23 @@ func updateReleasesFile(_ *cobra.Command, _ []string) (retErr error) {
 	if err := validateReleaseData(result); err != nil {
 		return fmt.Errorf("failed to validate downloaded data: %w", err)
 	}
-	addCurrentRelease(result)
+	currentVersion := version.MustParse(build.BinaryVersion())
+	addCurrentRelease(result, currentVersion)
 
 	fmt.Printf("writing results to %s\n", releaseDataFile)
-	if err := saveResults(result); err != nil {
+	if err := saveResultsInYaml(result); err != nil {
+		return err
+	}
+	key := fmt.Sprintf("%d.%d", currentVersion.Major(), currentVersion.Minor())
+	predecessor := result[key].Latest
```

Comment:
Nit: `currentSeries` is probably a better name for `key`.

Shouldn't this always be `result[result[key].Predecessor].Latest`?

For example, when we release the first rc or beta for 23.2, the `23.2` entry in the releases YAML will have a `Latest` field. However, if the `version.txt` file still indicates that master is `23.2`, we should still be using the latest 23.1 as predecessor. We should only start using a 23.2 release as predecessor once `version.txt` (and hence `currentVersion`) is bumped.

Reply:
> Shouldn't this always be result[result[key].Predecessor].Latest?

I have no idea. If you believe this is what it should be then I'm inclined to think you're right.